### ISSUE TITLE
Build issue with gcc 12 on Fedora 36 beta

### ISSUE THREAD

**Issue by ycollet** [ossia/score#1392]
I tried to build Ossia last release on Fedora 36 Beta using gcc-12.
The build failed with the following error:
```
/builddir/build/BUILD/score-v3.0.7/3rdparty/vst3/pluginterfaces/base/funknown.cpp: In function 'Steinberg::int32 Steinberg::FUnknownPrivate::atomicAdd(Steinberg::int32&, Steinberg::int32)':
/builddir/build/BUILD/score-v3.0.7/3rdparty/vst3/pluginterfaces/base/funknown.cpp:92:51: error: 'atomic_int_least32_t' does not name a type
   92 |         return atomic_fetch_add (reinterpret_cast<atomic_int_least32_t*> (&var), d) + d;
      |                                                   ^~~~~~~~~~~~~~~~~~~~
/builddir/build/BUILD/score-v3.0.7/3rdparty/vst3/pluginterfaces/base/funknown.cpp:92:71: error: expected '>' before '*' token
   92 |         return atomic_fetch_add (reinterpret_cast<atomic_int_least32_t*> (&var), d) + d;
      |                                                                       ^
/builddir/build/BUILD/score-v3.0.7/3rdparty/vst3/pluginterfaces/base/funknown.cpp:92:71: error: expected '(' before '*' token
   92 |         return atomic_fetch_add (reinterpret_cast<atomic_int_least32_t*> (&var), d) + d;
      |                                                                       ^
      |                                                                       (
/builddir/build/BUILD/score-v3.0.7/3rdparty/vst3/pluginterfaces/base/funknown.cpp:92:72: error: expected primary-expression before '>' token
   92 |         return atomic_fetch_add (reinterpret_cast<atomic_int_least32_t*> (&var), d) + d;
      |                                                                        ^
gmake[2]: *** [vst3/pluginterfaces/CMakeFiles/pluginterfaces.dir/build.make:107: vst3/pluginterfaces/CMakeFiles/pluginterfaces.dir/base/funknown.cpp.o] Error 1
gmake[2]: Leaving directory '/builddir/build/BUILD/score-v3.0.7/redhat-linux-build'
gmake[1]: *** [CMakeFiles/Makefile2:2924: vst3/pluginterfaces/CMakeFiles/pluginterfaces.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
```

Probably a missing header ...

**Comment by jcelerier** [comment]
ah damn, that's an issue with GCC 12 and the steinberg SDK ; there's been similar issues on windows: https://forums.steinberg.net/t/pluginterfaces-lib-compilation-error-win-10-vs-2022/768976

**Comment by jcelerier** [comment]
I'll force  -DSMTG_USE_STDATOMIC_H=OFF on the ossia side

**Comment by jcelerier** [comment]
by the way, you think that you could provide a CI script so that we can add it to the tested platforms and catch these issues automatically ? 

It's mainly: 

- Adding it in this file: (there's already a fedora 33 which I guess can be 90% copy-pasted)

https://github.com/ossia/score/blob/master/.github/workflows/builds.yaml

- If it needs specific scripts different from fedora 33 then adding them in this folder following the existing fedora ones would be the way to go: 

https://github.com/ossia/score/blob/master/ci/fedora.deps.sh

https://github.com/ossia/score/blob/master/ci/fedora.build.sh

e.g. fedora36.{deps,build}.sh following the existing pattern

**Comment by ycollet** [comment]
Yes, thanks, I will try that. I fork your repo and will propose a PR ...

**Comment by ycollet** [comment]
I reopen the ticket for a new one. It's nearly just before the linking phase so, I hope it will be the last one :)
¹¹¹
In file included from /builddir/build/BUILD/score-v3.0.8/redhat-linux-build/src/plugins/score-plugin-avnd/CMakeFiles/score_plugin_avnd.dir/cmake_pch.hxx:12,
                 from <command-line>:
/builddir/build/BUILD/score-v3.0.8/src/plugins/score-plugin-avnd/Crousti/Layer.hpp: In member function 'QGraphicsItem* oscr::LayoutBuilder<Info>::createCustom(T&)':
/builddir/build/BUILD/score-v3.0.8/src/plugins/score-plugin-avnd/Crousti/Layer.hpp:64:52: error: expected ';' before '{' token
   64 |       if constexpr(requires { typename T::item_type{}; }) {
      |                                                    ^
/builddir/build/BUILD/score-v3.0.8/src/plugins/score-plugin-avnd/Crousti/Layer.hpp: In member function 'auto oscr::LayerFactory<Info>::makeItemImpl(oscr::ProcessModel<Info>&, QGraphicsItem*) const':
/builddir/build/BUILD/score-v3.0.8/src/plugins/score-plugin-avnd/Crousti/Layer.hpp:234:51: error: expected ';' before '{' token
  234 |     if constexpr(requires { typename Info::ui::bus{}; }) {
      |                                                   ^
[ 95%] Building CXX object src/plugins/score-plugin-controlsurface/CMakeFiles/score_plugin_controlsurface.dir/score_plugin_controlsurface.cpp.o
¹¹¹

**Comment by jcelerier** [comment]
I think it should be good now - we have some GCC 12 CI which runs fine
